### PR TITLE
Correct handing of signals by the application

### PR DESCRIPTION
**Pull request by AdamAkiva**
This is tagged as BE since it is mostly relevant for BE but the changes were made for FE as well.

Basically there was an issue with forwarding signals to the docker images, you saw it when the `./remove.sh` script reached the timeout of 10 seconds every time until the flag `stop_grace_period` was added.
However, this avoided the issue, the issue was that the signals never reached the node application inside the docker.
Links for reference and research:
https://stackoverflow.com/questions/34441975/sigterm-does-not-reach-node-script-when-docker-runs-it-with-bin-sh-c
https://hynek.me/articles/docker-signals/
https://emmer.dev/blog/docker-shell-vs.-exec-form/

What was done was basically to add [tini](https://github.com/krallin/tini) as the entrypoint of the images, and in order to allow for multiple commands to run a script was added for each.
I was unable to use `npm` or `&&` since both of them create a new sub-shell which caused `tini` to not forward the signals to the correct PID.

In general there are not many changes but it is an advanced and really exotic concept for developers (very important for devops and deployment orchestration tools/admins, at least in my opinion)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM node:20.11.1-alpine as dev-base
```
Comment (Meital-Kozhidov) [inline Dockerfile:1]: What is the motivation for one Dockerfile in root?

File: docker-compose.yml [hunk 2]
```diff
@@ -11,13 +11,12 @@ services:
       - POSTGRES_DB=dashboard-dev
```
Comment (Meital-Kozhidov) [inline docker-compose.yml:11]: Why the change to root?

File: docker-compose.yml [hunk 3]
```diff
@@ -11,13 +11,12 @@ services:
...
     volumes:
       - ${DB_INIT_SCRIPT}:/docker-entrypoint-initdb.d/init.sql:ro
-      - ../db-dev-data/pg:/var/lib/postgresql/data:rw
+      - ./db-dev-data/pg:/var/lib/postgresql/data:rw
       # See: https://github.com/docker-library/docs/blob/master/postgres/README.md#arbitrary---user-notes
       - /etc/passwd:/etc/passwd:ro
       # See: https://stackoverflow.com/a/44440563
       - /etc/timezone:/etc/timezone:ro
       - /etc/localtime:/etc/localtime:ro
-    stop_grace_period: 2s
```
Comment (Meital-Kozhidov) [inline docker-compose.yml:20]: Why is this line removed everywhere?

File: scripts/be-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+# This script is used for the docker start up. It sets up all of the required
+# dependencies and start the process as PID 1 as a result of running this script
+# using tini. This will allow signals to be forwarded to the application as
+# expected
+
+npm install && npm run generate:openapi;
+exec env DEBUG=dashboard:* ./node_modules/.bin/nodemon --config ./nodemon.json ./src/main.ts;
```
Comment (Meital-Kozhidov) [inline scripts/be-entrypoint.sh:9]: Why not as an `npm` script?

File: docker-compose.yml (hunk 3 above)
Comment (Meital-Kozhidov) [reply docker-compose.yml:20]: So.. it is just redundant now?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+RUN apk update && apk add --no-cache curl=8.5.0-r0 && apk add --no-cache tini=0.19.0-r2
+
+############################### Development BE ##################################
+
+FROM dev-base as be-dev
+
+# Set the workdir
+WORKDIR /home/node/dashboard
+
+# Copy the entrypoint script to a non-volumed folder
```
Comment (Meital-Kozhidov) [inline Dockerfile:13]: What's the difference?

Comment (AdamAkiva) [reply Dockerfile:13]: I don't quite understand the question, so I'll just say what it does.
The idea here was to copy the `entrypoint.sh` script to non-volumed place.
The main reason for it was a fear of (never checked it) undefined behavior when changing the `entrypoint.sh` script while it was used to run the application.
I don't know whether it affects the underlying docker so I avoided it instead of figuring a way to check for it

File: Dockerfile (hunk 1 above)
Comment (Meital-Kozhidov) [reply Dockerfile:1]: And it will also be the prod docker? isn't it too verbose for one file?

File: docker-compose.yml (hunk 2 above)
Comment (Meital-Kozhidov) [reply docker-compose.yml:11]: Why 1 is a problem if this file is inside the scripts folder?

File: Dockerfile (hunk 5 above)
Comment (Meital-Kozhidov) [reply Dockerfile:13]: The question was about the non-volumed place, so you answered it. I would expect it not to effect the underlying docker anyway, but I get the avoidance instead

File: Dockerfile (hunk 1 above)
Comment (AdamAkiva) [reply Dockerfile:1]: I don't believe it is.
I do believe a single file is better, what are your arguments against it?

File: Dockerfile (hunk 5 above)
Comment (AdamAkiva) [reply Dockerfile:13]: I assume so as well.
I may check it just to be sure at some point

File: docker-compose.yml (hunk 2 above)
Comment (AdamAkiva) [reply docker-compose.yml:11]: Shouldn't be an issue, but there's no need for it.
According to the [PoLP](https://en.wikipedia.org/wiki/Principle_of_least_privilege) if there's no need than why should we?

File: Dockerfile (hunk 1 above)
Comment (Meital-Kozhidov) [reply Dockerfile:1]: Same reasons I would split to multiple files a file that's too long; Comment lines for separation is great, but too long files with multiple themes might be hard to navigate through. I'm not against it, just trying to understand your prespective

File: docker-compose.yml (hunk 2 above)
Comment (Meital-Kozhidov) [reply docker-compose.yml:11]: Okay, I just wanted to understand

File: Dockerfile (hunk 1 above)
Comment (AdamAkiva) [reply Dockerfile:1]: When working with multiple docker files, change in one usually implicates a change in another unless we missed a bug.
A node version update is a good example for it.
I agree in general, however since most changes are cross-wide, and the fact that multiple docker files will require different names or locations, I prefer this approach

File: docker-compose.yml (hunk 2 above)
Comment (AdamAkiva) [reply docker-compose.yml:11]: And I hope I explained it well enough

File: Dockerfile (hunk 1 above)
Comment (Meital-Kozhidov) [reply Dockerfile:1]: I'll accept that
